mixer/pkg/runtime/dispatcher/dispatchstate.go

```diff
@@ -79,9 +79,26 @@ func (ds *dispatchState) beginSpan(ctx context.Context) (opentracing.Span, conte
 	return span, ctx, time.Now()
 }
 
+func (ds *dispatchState) logCheckResultToDispatchSpan(span opentracing.Span) {
+	if ds.session.impl.enableTracing {
+		logCheckResultToDispatchSpan(span, ds.destination.Template.Name, ds.destination.HandlerName, ds.destination.AdapterName, ds.checkResult, ds.err)
+	}
+}
+
+func (ds *dispatchState) logQuotaResultToDispatchSpan(span opentracing.Span) {
+	if ds.session.impl.enableTracing {
```

Comment:
You mean it could be possible that the value `ds.session.impl.enableTracing` changes between the 2 consecutive calls in `loqQuotaResultToDispatchSpan()` and `completeSpan()`? 
I can't currently imagine how this should happen unless the fields are mutated concurrently, but I sincerely hope that this doesn't happen.

Reply:
@mandarjog To me it looks like sampling when the span is finished:
https://github.com/istio/istio/blob/8b930144c13ff16b6fced36fbb2b29a85119f0e1/mixer/pkg/runtime/dispatcher/dispatchstate.go#L85
And then eventually here:
https://github.com/istio/istio/blob/8b930144c13ff16b6fced36fbb2b29a85119f0e1/vendor/github.com/uber/jaeger-client-go/span.go#L188

`ds.session.impl.enableTracing` just seems to constant at runtime and `true` if mixer was started with a jaeger or zipkin url or if it should log the spans.